lookup_plugins/oo_option.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python2
+# -*- coding: utf-8 -*-
+# vim: expandtab:tabstop=4:shiftwidth=4
+
+'''
+oo_option lookup plugin for openshift-ansible
+
+Usage:
+
+    - debug:
+      msg: "{{ lookup('oo_option', '<key> default=<default_value>') }}"
```

Comment:
@detiber I must be missing something. In the first example, wouldn't 'name' be set to an empty string?

And isn't that exactly the behavior we'd be looking for?

My understanding is that inside of the filter pipeline, all filters get the data and have a chance to mutate it and pass it along.

So, in the first example, the lookup plugin looks up the value (in this case empty string), and passes that to the default filter.

Then default filter takes that value and determines if it should change it. In this case, since we passed in 'true' as the second parameter, empty strings do get replaced.

So, wouldn't 'name' end up being the value 'CentOS-7-x86_64-GenericCloud.qcow2' ?

I agree that in the second example, 'name' would be set to an empty string, but again, that's what we'd expect. I think I'm totally missing your point.